packages/eslint-plugin-template/src/rules/self-closing-tags.ts

```diff
@@ -0,0 +1,74 @@
+import type {
+  TmplAstElement,
+  TmplAstText,
+} from '@angular-eslint/bundled-angular-compiler';
+import {
+  getTemplateParserServices,
+  NATIVE_ELEMENTS,
+} from '@angular-eslint/utils';
+import { createESLintRule } from '../utils/create-eslint-rule';
+
+export const MESSAGE_ID = 'selfClosingTags';
+export const RULE_NAME = 'self-closing-tags';
+
+export default createESLintRule<[], typeof MESSAGE_ID>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Ensures that self-closing tags are used for elements with a closing tag but no content.',
+      recommended: false,
+    },
+    fixable: 'code',
+    schema: [],
+    messages: {
+      [MESSAGE_ID]:
+        'Use self-closing tags for elements with a closing tag but no content.',
+    },
+  },
+  defaultOptions: [],
+  create(context) {
+    const parserServices = getTemplateParserServices(context);
+
+    return {
+      Element$1({
```

Comment:
Please see comment about applying the filter of non-native elements in the selector